File: twentyfive/src/test/java/uk/chaoticgoose/adventofcode/twentyfive/day5/Day5Test.java
package uk.chaoticgoose.adventofcode.twentyfive.day5;

import org.junit.jupiter.api.Test;
import uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader;
import uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier.LONG;
import static uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier.SHORT;

class Day5Test {
    private final TestInputLoader testInputLoader = new TestInputLoader();
    private final InputParser inputParser = new InputParser();
    private final DaySolution underTest = new DaySolution();

    @Test
    void part1Short() {
        assertThat(underTest.part1(inputParser.parse(data(SHORT)))).isEqualTo(3L);
    }

    @Test
    void part1Long() {
        assertThat(underTest.part1(inputParser.parse(data(LONG)))).isEqualTo(758L);
    }

    @Test
    void part2Short() {
        assertThat(underTest.part2(inputParser.parse(data(SHORT)))).isEqualTo(14L);
    }

    @Test
    void part2Long() {
        assertThat(underTest.part2(inputParser.parse(data(LONG)))).isEqualTo(343143696885053L);
    }

    private List<String> data(InputDataModifier inputDataModifier) {
        return testInputLoader.load(5, inputDataModifier);
    }
}

File: src/test/kotlin/twenty_four/day5/Day5Test.kt
package twenty_four.day5

import assertk.assertThat
import assertk.assertions.isEqualTo
import loadData
import org.junit.jupiter.api.Test

class Day5Test {
    private val parser = InputParser()

    @Test
    fun `part 1 long`() {
        assertThat(part1(parser.parse(data()))).isEqualTo(7024)
    }

    @Test
    fun `part 1 short`() {
        assertThat(part1(parser.parse(shortData()))).isEqualTo(143)
    }

    @Test
    fun `part 2 long`() {
        assertThat(part2(parser.parse(data()))).isEqualTo(4151)
    }

    @Test
    fun `part 2 short`() {
        assertThat(part2(parser.parse(shortData()))).isEqualTo(123)
    }

    private fun data(modifier: String? = null) = loadData(2024, 5, modifier)

    private fun shortData() = data("short")
}
